<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1014</width>
    <height>758</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>hal-core-2.0</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(51, 57, 59);
color: rgb(255, 255, 255);
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(51, 57, 59);
color: rgb(255, 255, 255);
</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="0">
     <widget class="QFrame" name="frame">
      <property name="maximumSize">
       <size>
        <width>275</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="7" column="2">
          <widget class="QPushButton" name="pushButton_j2_plus">
           <property name="text">
            <string>j2+</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QSpinBox" name="spinBox_stepsize_joint">
           <property name="minimum">
            <number>-1000</number>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item row="17" column="1">
          <widget class="QPushButton" name="pushButton_tool_y_min">
           <property name="text">
            <string>tool y-</string>
           </property>
          </widget>
         </item>
         <item row="18" column="0">
          <widget class="QCheckBox" name="checkBox_ik">
           <property name="text">
            <string>Ik</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="17" column="0">
          <widget class="QCheckBox" name="checkBox_fk">
           <property name="text">
            <string>Fk</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLineEdit" name="lineEdit_cart_z">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLineEdit" name="lineEdit_j2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QPushButton" name="pushButton_j3_min">
           <property name="text">
            <string>j3-</string>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QPushButton" name="pushButton_z_plus">
           <property name="text">
            <string>z+</string>
           </property>
          </widget>
         </item>
         <item row="5" column="2">
          <widget class="QPushButton" name="pushButton_j0_plus">
           <property name="text">
            <string>j0+</string>
           </property>
          </widget>
         </item>
         <item row="14" column="1">
          <widget class="QPushButton" name="pushButton_euler_z_min">
           <property name="text">
            <string>euler z-</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="pushButton_x_plus">
           <property name="text">
            <string>x+</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QPushButton" name="pushButton_y_plus">
           <property name="text">
            <string>y+</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QPushButton" name="pushButton_j0_min">
           <property name="text">
            <string>j0-</string>
           </property>
          </widget>
         </item>
         <item row="18" column="2">
          <widget class="QPushButton" name="pushButton_tool_z_plus">
           <property name="text">
            <string>tool z+</string>
           </property>
          </widget>
         </item>
         <item row="12" column="1">
          <widget class="QPushButton" name="pushButton_euler_x_min">
           <property name="text">
            <string>euler x-</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLineEdit" name="lineEdit_cart_y">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLineEdit" name="lineEdit_j0">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="19" column="0" colspan="3">
          <widget class="QCheckBox" name="checkBox_ik_from_init">
           <property name="toolTip">
            <string>ik from init position 
or ik from current position.</string>
           </property>
           <property name="text">
            <string>Ik from init</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1" colspan="2">
          <widget class="QLabel" name="label_stepsize_cart">
           <property name="text">
            <string>stepsize cart [mm]</string>
           </property>
          </widget>
         </item>
         <item row="13" column="1">
          <widget class="QPushButton" name="pushButton_euler_y_min">
           <property name="text">
            <string>euler y-</string>
           </property>
          </widget>
         </item>
         <item row="9" column="0">
          <widget class="QLineEdit" name="lineEdit_j4">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLineEdit" name="lineEdit_j3">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="12" column="0">
          <widget class="QLineEdit" name="lineEdit_euler_x">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="10" column="0">
          <widget class="QLineEdit" name="lineEdit_j5">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QPushButton" name="pushButton_z_min">
           <property name="text">
            <string>z-</string>
           </property>
          </widget>
         </item>
         <item row="10" column="2">
          <widget class="QPushButton" name="pushButton_j5_plus">
           <property name="text">
            <string>j5+</string>
           </property>
          </widget>
         </item>
         <item row="11" column="0">
          <widget class="QSpinBox" name="spinBox_stepsize_euler">
           <property name="minimum">
            <number>-1000</number>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="pushButton_y_min">
           <property name="text">
            <string>y-</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QPushButton" name="pushButton_j1_min">
           <property name="text">
            <string>j1-</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLineEdit" name="lineEdit_cart_x">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="13" column="0">
          <widget class="QLineEdit" name="lineEdit_euler_y">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="9" column="1">
          <widget class="QPushButton" name="pushButton_j4_min">
           <property name="text">
            <string>j4-</string>
           </property>
          </widget>
         </item>
         <item row="6" column="2">
          <widget class="QPushButton" name="pushButton_j1_plus">
           <property name="text">
            <string>j1+</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="pushButton_x_min">
           <property name="text">
            <string>x-</string>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLineEdit" name="lineEdit_j1">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="15" column="1" colspan="2">
          <widget class="QLabel" name="label_stepsize_tooldir">
           <property name="text">
            <string>stepsize tooldir [mm]</string>
           </property>
          </widget>
         </item>
         <item row="14" column="0">
          <widget class="QLineEdit" name="lineEdit_euler_z">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QSpinBox" name="spinBox_stepsize_cart">
           <property name="minimum">
            <number>-1000</number>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item row="4" column="1" colspan="2">
          <widget class="QLabel" name="label_stepsize_joint">
           <property name="text">
            <string>stepsize joint [deg]</string>
           </property>
          </widget>
         </item>
         <item row="15" column="0">
          <widget class="QSpinBox" name="spinBox_stepsize_tooldir">
           <property name="minimum">
            <number>-1000</number>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item row="12" column="2">
          <widget class="QPushButton" name="pushButton_euler_x_plus">
           <property name="text">
            <string>euler x+</string>
           </property>
          </widget>
         </item>
         <item row="14" column="2">
          <widget class="QPushButton" name="pushButton_euler_z_plus">
           <property name="text">
            <string>euler z+</string>
           </property>
          </widget>
         </item>
         <item row="18" column="1">
          <widget class="QPushButton" name="pushButton_tool_z_min">
           <property name="text">
            <string>tool z-</string>
           </property>
          </widget>
         </item>
         <item row="8" column="2">
          <widget class="QPushButton" name="pushButton_j3_plus">
           <property name="text">
            <string>j3+</string>
           </property>
          </widget>
         </item>
         <item row="16" column="2">
          <widget class="QPushButton" name="pushButton_tool_x_plus">
           <property name="text">
            <string>tool x+</string>
           </property>
          </widget>
         </item>
         <item row="11" column="1" colspan="2">
          <widget class="QLabel" name="label_stepsize_euler">
           <property name="text">
            <string>stepsize euler [deg]</string>
           </property>
          </widget>
         </item>
         <item row="9" column="2">
          <widget class="QPushButton" name="pushButton_j4_plus">
           <property name="text">
            <string>j4+</string>
           </property>
          </widget>
         </item>
         <item row="16" column="0">
          <widget class="QCheckBox" name="checkBox_error">
           <property name="text">
            <string>Ik error</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="13" column="2">
          <widget class="QPushButton" name="pushButton_euler_y_plus">
           <property name="text">
            <string>euler y+</string>
           </property>
          </widget>
         </item>
         <item row="20" column="0">
          <widget class="QSpinBox" name="spinBox_ik_iterations">
           <property name="value">
            <number>50</number>
           </property>
          </widget>
         </item>
         <item row="16" column="1">
          <widget class="QPushButton" name="pushButton_tool_x_min">
           <property name="text">
            <string>tool x-</string>
           </property>
          </widget>
         </item>
         <item row="10" column="1">
          <widget class="QPushButton" name="pushButton_j5_min">
           <property name="text">
            <string>j5-</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QPushButton" name="pushButton_j2_min">
           <property name="text">
            <string>j2-</string>
           </property>
          </widget>
         </item>
         <item row="17" column="2">
          <widget class="QPushButton" name="pushButton_tool_y_plus">
           <property name="text">
            <string>tool y+</string>
           </property>
          </widget>
         </item>
         <item row="20" column="1" colspan="2">
          <widget class="QLabel" name="label_iteriations">
           <property name="text">
            <string>ik iterations [n]</string>
           </property>
          </widget>
         </item>
         <item row="21" column="0">
          <widget class="QPushButton" name="pushButton_reset">
           <property name="text">
            <string>reset</string>
           </property>
          </widget>
         </item>
         <item row="21" column="1" colspan="2">
          <widget class="QCheckBox" name="checkBox_motion_continious">
           <property name="text">
            <string>Motion continious</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QFrame" name="frame_2">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout_opencascade"/>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1014</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
